src/app/model/ChessEngine.ts

```diff
@@ -64,6 +65,44 @@ export class ChessEngine implements ChessModel {
     return [pieceWasCaptured, pieceWasMoved, kingWasChecked, kingWasUnchecked, pawnPromotionWasEnabled].filter(this.hasOccurred);
   }
 
+  pawnWasPromoted(chosenPiece: string): PawnWasPromoted | undefined {
+    if (this.promotingOnSquare) {
```

Comment:
Im mniej kodu jest w zagniezdzeniu - tym jest czytelniej :) Wiec mozna to odwrocic:
```
if (!this.promotingOnSquare) {
 return undefined;
}
      const onSquarePromotion = this.promotingOnSquare;
      const pawnToRemove = this.squaresWithPiece[this.translateSquareToAlgebraicNotationCapital(onSquarePromotion)];
      const createChosenPiece = this.objectCreator(chosenPiece, pawnToRemove.side);
      this.squaresWithPiece[this.translateSquareToAlgebraicNotationCapital(onSquarePromotion)] = createChosenPiece;
      this.promotingOnSquare = undefined;
      this.currentSide = this.anotherSide(this.currentSide);

      return {
        eventType: 'PawnWasPromoted',
        onSquare: onSquarePromotion,
        chosenPiece: createChosenPiece,
      };
```